test/test.downscopedclient.ts

```diff
@@ -374,6 +374,126 @@ describe('DownscopedClient', () => {
       );
       scope.done();
     });
+
+    it('should copy source cred expiry time if STS response does not return expiry time', async () => {
```

Comment:
This test confuses me. Based on the description, all we should be doing is (which you are doing):
1. Initialize a source credential with a certain expiration time
2. Mock STS to not return expires_in
3. Initialize DownscopedClient with the source credential
4. Call getAccessToken, and assert that the expiration matches the source credential

I don't think you need mock the clock so that the credentials you set expire. If you want to test that path, that can be a separate test?